### PR TITLE
Spanner Batch table

### PR DESCRIPTION
**Pull request by gdbelvin**

### CODE REVIEW

File: impl/spanner/batch/batch_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		if err != nil {
+			t.Fatalf("ReadBatch(%v): %v", tc.rev, err)
+		}
+		if !cmp.Equal(got, tc.want, cmp.Comparer(proto.Equal)) {
+			t.Errorf("ReadBatch(%v): %v, want %v", tc.rev, got, tc.want)
+		}
+	}
+}
+
+func TestHightestRev(t *testing.T) {
```
Comment (mhutchinson) [inline impl/spanner/batch/batch_test.go:131]: sp: s/Hightest/Highest/ unless pun is intentional. In which case, well played.

File: impl/spanner/batch/batch_test.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	} {
+		err := m.WriteBatchSources(ctx, directoryID, tc.rev, &spb.MapMetadata{Sources: tc.sources})
+		if got, want := err != nil, tc.wantErr; got != want {
+			t.Errorf("WriteBatchSources(%v, %v): err: %v. code: %v, want %v",
+				tc.rev, tc.sources, err, got, want)
+		}
+	}
+}
+
+func TestReadBatch(t *testing.T) {
```
Comment (mhutchinson) [inline impl/spanner/batch/batch_test.go:99]: Add coverage for reading a revision which doesn't yet exist?

File: impl/spanner/batch/batch.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+type Table struct {
+	client *spanner.Client
+}
+
+// New returns a new Table.
+func New(client *spanner.Client) *Table {
+	return &Table{client: client}
+}
+
+// WriteBatchSources saves the (low, high] boundaries used for each log in making this revision.
```
[the remark this answers is not in the document]
Comment (gdbelvin) [reply impl/spanner/batch/batch.go:40]: Sounds good.

File: impl/spanner/batch/batch_test.go (hunk 1 above)
Comment (gdbelvin) [reply impl/spanner/batch/batch_test.go:131]: Thanks for the catch

File: impl/spanner/batch/batch_test.go (hunk 2 above)
Comment (gdbelvin) [reply impl/spanner/batch/batch_test.go:101]: done

**Comment by gdbelvin** [comment]
Thanks for taking the time to look at it. Updated with nits. PTAL

**Pull request merged**